wrangler-core/src/main/java/io/cdap/directives/column/Keep.java

```diff
@@ -93,4 +103,24 @@ public Mutation lineage() {
     keep.forEach(column -> builder.relation(column, column));
     return builder.build();
   }
+  @Override
+  public Relation transform(RelationalTranformContext relationalTranformContext,
+                            Relation relation) {
+    Optional<ExpressionFactory<String>> expressionFactory = getExpressionFactory(relationalTranformContext);
+    if (!expressionFactory.isPresent()) {
+      return new InvalidRelation("Cannot find an Expression Factory");
+    }
+    Map<String, Expression> keepCol = generateColumnExpMap(keep.stream().collect(Collectors.toList()),
+            expressionFactory.get());
+    return relation.select(keepCol);
+  }
+
+  private Optional<ExpressionFactory<String>> getExpressionFactory(RelationalTranformContext ctx) {
+    return ctx.getEngine().getExpressionFactory(StringExpressionFactoryType.SQL);
+  }
+  private Map<String, Expression> generateColumnExpMap(List<String> columns, ExpressionFactory<String> factory) {
```

Comment:
Even this function seems like a util function rather than directive specific function. can we move this to util class or `Directive` interface.

Reply:
Yes. Will move it to a separate file.